test/unit/canvas.js

```diff
@@ -1290,6 +1290,25 @@
     assert.equal(canvas.toObject().objects[0].type, rect.type);
   });
 
+
+  QUnit.test('toObject with clipPath', function(assert) {
+    var clipPath = makeRect();
+    var canvasWithClipPath = new fabric.Canvas(null, { clipPath: clipPath });
+    var expectedObject = {
+      'version': fabric.version,
+      objects: canvasWithClipPath.getObjects(),
+      'clipPath': clipPath.toObject()
+    };
+
+    assert.ok(typeof canvasWithClipPath.toObject === 'function');
+    assert.deepEqual(expectedObject, canvasWithClipPath.toObject());
+
+    var rect = makeRect();
+    canvasWithClipPath.add(rect);
+
+    assert.equal(canvasWithClipPath.toObject().objects[0].type, rect.type);
```

Comment:
to be a test that catch regressions, this last assert is not important.
you have to dump the output of a canvas `toObject()` and copy paste it in the test.
That is fixed data and we check that the code respect the fixed data ( apart for the version that is not fixed ).

So without objects, you get an empty array, and that is fine.

```
    var expectedObject = {
      'version': fabric.version,
      objects: [],
      'clipPath': { type: 'rect', width.... }
    };
```